java/arcs/android/sdk/host/AndroidResurrector.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.sdk.host
+
+import android.app.Service
+import arcs.core.host.Resurrector
+import arcs.core.host.ResurrectorBase
+import arcs.core.storage.StorageKey
+import arcs.sdk.android.storage.ResurrectionHelper
+
+/**
+ * An implementation of [Resurrector] that is backed by [AndroidResurrector]. This implementation
+ * currently has limitations, in that it does not persist the arcId to storageKey mappings, but
+ * uses memory resident maps. This is a workaround until [ResurrectorService] is upgraded to
+ * support this feature.
+ */
+open class AndroidResurrector(val context: Service) : ResurrectorBase() {
```

Comment:
I loosened it to a Context, but keep in mind, this is used by ResurrectionService as the ComponentName to send a reply on, and normally by convention this would be the Service that's embedding the ArcHost

Reply:
Ok, I need to rethink this. When the ResurrectionHelper invokes our (potentially rebooted ArcHost), the onResurrected callback that ResurrectorBase is holding could be to an old stale instance of ArcHost, the callback always needs to run with the current instance.

The original intent of this was to have an abstracted version of ResurrectionHelper that can live in core, but it actually requires more than that, because there's no notion of 'onStartCommand' for core in the sense of "create a fresh instance of ArcHost, and then invoke this operation" like the way it works with Android. 

This is too much trouble to support something which doesn't benefit core (until some xplatform notion of something that can start archosts exists) at the moment, so I'm going to nuke the resurrector stuff and just have AndroidHost directly use ResurrectionHelper for now.